public/app/features/dashboard/components/AnnotationSettings/AnnotationSettingsEdit.tsx

```diff
@@ -142,6 +142,7 @@ export const AnnotationSettingsEdit = ({ editIdx, dashboard }: Props) => {
   const panels: Array<SelectableValue<number>> = useMemo(
     () =>
       dashboard?.panels
+        .filter((panel) => config.panels[panel.type])
```

Comment:
```suggestion
        .filter((panel) => panel.plugin?.dataSupport?.annotations)
```

@ryantxu @adela-almasan that should do it

Reply:
hm.,. the only problem with that is that it won't work for lazy loaded panels that have yet to to render (and load it's plugin), and this code does not take rows into account.

To get all panels including those in collapsed rows it's best to use dashboard.panelIterator,  but to look up if the visualization actually supports annotations is impossible without loading the PanelPlugin